server/services/offendersService.js

```diff
@@ -145,31 +144,44 @@ module.exports = function createOffendersService(nomisClientBuilder, formService
     try {
       const nomisClient = nomisClientBuilder(token)
       const uncategorisedResult = (await nomisClient.getUncategorisedOffenders(agencyId)).filter(
-        s => s.status === Status.AWAITING_APPROVAL.name
+        s => s.status === Status.AWAITING_APPROVAL.name // the status coming back from nomis
       )
 
-      if (isNilOrEmpty(uncategorisedResult)) {
+      const unapprovedWithDbRecord = await Promise.all(
+        uncategorisedResult.map(async s => {
+          const dbRecord = await formService.getCategorisationRecord(s.bookingId)
+          return {
+            ...s,
+            dbRecord,
+          }
+        })
+      )
+
+      // remove any sent back to categoriser records
+      const unapprovedOffenders = unapprovedWithDbRecord.filter(o => o.dbRecord.status !== Status.SUPERVISOR_BACK.name)
+
+      if (isNilOrEmpty(unapprovedOffenders)) {
         logger.info(`No unapproved offenders found for ${agencyId}`)
         return []
       }
 
       const sentenceMap = await getSentenceMap(uncategorisedResult, nomisClient)
 
       const decoratedResults = await Promise.all(
-        uncategorisedResult
-          .filter(o => sentenceMap.find(s => s.bookingId === o.bookingId)) // filter out offenders without sentence
-          .map(async o => {
-            const dbRecord = await formService.getCategorisationRecord(o.bookingId)
-            return {
-              ...o,
-              displayName: `${properCaseName(o.lastName)}, ${properCaseName(o.firstName)}`,
-              categoriserDisplayName: `${properCaseName(o.categoriserFirstName)} ${properCaseName(
-                o.categoriserLastName
-              )}`,
-              dbRecordExists: !!dbRecord.booking_id,
-              ...buildSentenceData(sentenceMap.find(s => s.bookingId === o.bookingId).sentenceDate),
-            }
-          })
+        unapprovedOffenders.map(async o => {
+          // TODO: async probably now redundant ^^
```

Comment:
You are correct. You no longer need the async or the promise all. Just the unapprovedOffenders.map(o => would do it

Reply:
ok ill go for it !